samtranslator/internal/model/appsync.py

```diff
@@ -6,6 +6,8 @@
 from samtranslator.model.intrinsics import fnGetAtt
 from samtranslator.utils.types import Intrinsicable
 
+NONE_DATASOURCE_LITERALS = {"NONE", "None", "none"}
```

Comment:
Why do we support all this? I'd rather start minimal and strict, and I'd argue simple (otherwise documentation needs to specify all these). We can add conveniences if it's really an issue later.

Reply:
If the intention is to forbid "none" altogether in case-insensitive way, I'd make the intention clear with `lower()`/`upper()` instead.

Reply:
I was just about to suggest that. We can just reserve "none" case insensitive and that would be easier to document too.